turbopack/crates/turbopack-core/src/resolve/parse.rs

```diff
@@ -263,8 +271,22 @@ impl Request {
                 Request::Relative { path, .. } => {
                     path.push(item);
                 }
-                Request::Module { path, .. } => {
-                    path.push(item);
+                Request::Module { module, path, .. } => {
+                    if !was_module_name_terminated && matches!(item, Pattern::Dynamic) {
+                        // TODO ideally this would be more general (i.e. support also
+                        // `module-part<dynamic>more-module/subpath`) and not just handle
+                        // Pattern::Dynamic, but this covers the common case of
+                        // `require('@img/sharp-' + arch + '/sharp.node')`
+
+                        // Insert dynamic between module and path (by adding it to both of them,
+                        // because both could happen). Note that path is empty at this
+                        // point anyway.
+                        module.push(Pattern::DynamicNoSlash);
+                        path.push(item);
+                    } else {
+                        path.push(item);
+                        was_module_name_terminated = true;
+                    }
```

Comment:
```suggestion
                    if path.is_empty() && matches!(item, Pattern::Dynamic) {
                        // TODO ideally this would be more general (i.e. support also
                        // `module-part<dynamic>more-module/subpath`) and not just handle
                        // Pattern::Dynamic, but this covers the common case of
                        // `require('@img/sharp-' + arch + '/sharp.node')`

                        // Insert dynamic between module and path (by adding it to both of them,
                        // because both could happen). Note that path is empty at this
                        // point anyway.
                        module.push(Pattern::DynamicNoSlash);
                    }
                    path.push(item);
```

Reply:
Try to parse `module<dynamic>/path<dynamic>path